multiqc/modules/fastqc/fastqc.py

```diff
@@ -195,14 +195,31 @@ def parse_fastqc_report(self, file_contents, s_name=None, f=None):
             d["measure"]: d["value"] for d in self.fastqc_data[s_name]["basic_statistics"]
         }
 
+        # we sort by the avg of the range, which is effectively
+        # sorting ranges in asc order assuming no overlap
+        sequence_length_distributions = self.fastqc_data[s_name].get("sequence_length_distribution", [])
+        sequence_length_distributions.sort(key=lambda d: self.avg_bp_from_range(d["length"]))
+
         # Calculate the average sequence length (Basic Statistics gives a range)
+        length_reads = 0
         length_bp = 0
-        total_count = 0
-        for d in self.fastqc_data[s_name].get("sequence_length_distribution", {}):
+        total_count = sum(d["count"] for d in sequence_length_distributions)
+        median = None
+
+        for d in sequence_length_distributions:
+            length_reads += d["count"]
             length_bp += d["count"] * self.avg_bp_from_range(d["length"])
-            total_count += d["count"]
+
+            if median is None and length_reads >= total_count / 2:
+                # if the distribution-entry is a range, we use the average of the range.
+                # this isn't technically correct, because we can't know what the distribution
+                # is within that range. Probably good enough though.
```

Comment:
We should really ask @s-andrews to add mean / median read length as an output metric from FastQC directly instead of fudging it like this 🤔 

@jchorl - if you fancy adding an issue that would be nice 😉  https://github.com/s-andrews/FastQC

Reply:
Will do!